xbmc/pvr/PVRManager.cpp

```diff
@@ -627,15 +627,28 @@ bool CPVRManager::Load(void)
   /* start the add-on update thread */
   if (m_addons)
     m_addons->Start();
+  else
+    return false;
 
-  /* load at least one client */
-  while (IsInitialising() && m_addons && !m_addons->HasConnectedClients())
-    Sleep(50);
+  /* wait until the addons have loaded */
+  m_addons->WaitForInitialisation();
 
-  if (!IsInitialising() || !m_addons || !m_addons->HasConnectedClients())
-    return false;
+  int numConnectedClients = m_addons->ConnectedClientAmount();
+  int numEnabledClients = m_addons->EnabledClientAmount();
+
+  if (numConnectedClients == 0)
+  {
+    CLog::Log(LOGINFO, "PVRManager - %s - %d of %d enabled clients connected in time, continue to start",
+      __FUNCTION__, numConnectedClients, numEnabledClients);
+  }
+  else
+  {
+    CLog::Log(LOGINFO, "PVRManager - %s - %d of %d enabled clients connected in time, continue to start", 
+      __FUNCTION__, numConnectedClients, numEnabledClients);
+  }
```

Comment:
Not sure what currently is the status of this request, but is there something on "numConnectedClients" check missing? Is only the log message and the same on both places.